### ISSUE TITLE
Not recognizing Citizens

### ISSUE THREAD

**Issue by Curtis73** [FlyingPikachu/Quests#631]
**What is your Quests version / build number (do _not_ say "latest")?**
3.5.9-b586

**Describe the problem and how to replicate it. Any console errors to post _using Pastebin.com_?**
Quests is reporting that I don't have Citizens installed in the editor when I do.
![image](https://user-images.githubusercontent.com/14040165/51079779-d9cd7580-16c5-11e9-9b02-c5464ab689d9.png)

I'm running Citizens 2.0.24-SNAPSHOT (built 1605)
Console is not reporting any errors.

**If applicable, can you provide an example quest from quests.yml _using Pastebin.com_?**
N/A

**Comment by FlyingPikachu** [comment]
Post your latest startup log, typically located in your server's /logs folder.

**Comment by Curtis73** [comment]
https://pastebin.com/SDW2GhTF

**Comment by FlyingPikachu** [comment]
> [00:40:31] [Server thread/INFO]: [Magic] Loading Magic v7.6.20-SNAPSHOT-86275ac
[00:40:31] [Server thread/INFO]: [Magic] Found WorldGuard 7+

> [00:40:54] [Server thread/INFO]: [Quests] Enabling Quests v3.5.9-b586
[00:40:54] [Server thread/WARN]: [Quests] Citizens was detected, but is not enabled! Fix Citizens to allow linkage.
[00:40:54] [Server thread/WARN]: [Quests] WorldGuard was detected, but is not enabled! Fix WorldGuard to allow linkage.

> [00:40:58] [Server thread/INFO]: [Citizens] Enabling Citizens v2.0.24-SNAPSHOT (build 1605)

According to those timestamps, Quests is loading before Citizens and WorldGuard have been enabled. Since Magic currently _requires_ Quests be loaded before it, but also depends on Citizens and WorldGuard, I suspect it to be the issue. In layman's terms, we see the following order of loading:

1. Quests
2. Citizens/WorldGuard
3. Magic

When Quests actually expects to see:

1. Citizens/WorldGuard
2. Magic
3. Quests

I believe if Magic were to `soft-depend` Quests rather than `loadbefore` it, this would resolve the issue. However, we'll have to ask @NathanWolf whether I'm right and the change would break anything under the hood.

**Comment by NathanWolf** [comment]
I think I had to do the loadbefore to make my Quests module work, otherwise it tries to load stuff from Magic before it is loaded.

Kinda ugly though. If it’s causing problems I can see if there’s another way, but I think due to the nature of how modules load I can’t really delay the access - once Quests wants them loaded they have to do their thing, though I could be wrong.

**Comment by FlyingPikachu** [comment]
@NathanWolf I'm seeing that the MythicMobs module does the same thing... not sure why that's necessary either. Will look into this myself if you don't get the chance. Thanks!

**Comment by FlyingPikachu** [comment]
So I took Magic's plugin.yml and changed `soft-depend` to `loadbefore` without any negative effects. Load order is correct and I was able to complete a Craft Magic Wand objective without issue.

@Curtis73 Since Magic uses the MIT license, I've uploaded this version of Magic that you can use with Quests 3.5.9 to (seemingly) resolve the issue until the official plugin updates - https://www.dropbox.com/s/i7m0oo0e73ck3ll/Magic-7.6.19-edited.jar?dl=0

@NathanWolf You should be aware that the upcoming Quests 3.6.0 will cause the MagicQuests module to throw an IllegalAccessError, since Quester.currentQuests now uses getter/setter methods.

**Comment by Curtis73** [comment]
That's interesting, I just tried what you suggested but I can't see any difference, It's still saying that Citizens and WG aren't installed & the same errors are popping up on startup.

**Comment by Curtis73** [comment]
That's not done anything - not to worry though, I can wait a bit.
Thanks for your help!

**Comment by FlyingPikachu** [comment]
@Curtis73 Thanks for your patience! The new version of the MagicQuests module is out:

http://jenkins.elmakers.com/job/MagicQuests/95/

Make sure to try it with Quests 3.6.0+ and the modified Magic jar from earlier (if the modified one works, I'll suggest the actual plugin be updated as well). Let us know how it goes!

**Comment by Curtis73** [comment]
I'm not seeing any difference with that, sadly.
If you were able to get it to work, then perhaps it's a problem with my setup. If you want, I'll set up a localhost and get rid of my plugins until it does work

**Comment by SoraShiunin** [comment]
Hey I'm having sort of the same issue but its this instead:
07.02 21:12:18 [Server] INFO Enabling Quests v3.6.2-b646
07.02 21:12:18 [Server] WARN WorldGuard was detected, but is not enabled! Fix WorldGuard to allow linkage.
07.02 21:12:18 [Server] WARN Denizen was detected, but is not enabled! Fix Denizen to allow linkage.
07.02 21:12:18 [Server] WARN PhatLoots was detected, but is not enabled! Fix PhatLoots to allow linkage.
I have all 3 plugins and it worked on 1.12.2, I am now updating to 1.13.2 and it looks like they dont "link".

**Comment by Curtis73** [comment]
Well, there's nothing wrong with Magic or WorldGuard. After some investigating it appears that DiscordSRV is causing it, I can't see how though

**Comment by SoraShiunin** [comment]
I still got notification for this, Im just gonna add a small input;
DiscordSRV somehow made my server lag out and crash multiple times and sometimes broke other plugins in my server. Even tho it has nice features, it somehow interferes alot with what it shouldn't(their support also refused to help with the issue, could be different for you), there are alot of other discord plugins out there that can replace it.

**Comment by Curtis73** [comment]
That's funny, I've never really had any issues with DiscordSRV & unfortunately DiscordSRV's pretty important for my server so removing it isn't much of an option

**Comment by FlyingPikachu** [comment]
@Curtis73 Glad you were able to get it working! I'll tag @Scarsz in case they care to investigate this further.

**Comment by Scarsz** [comment]
@Curtis73 nothing in DiscordSRV prevents any other plugin from functioning in any way. It's open source. Go look through it if you want.

@SoraShiunin just looked through our support transcripts, nothing with your name on it. Nothing in DiscordSRV can make your server lag out, crash, nor break other plugins. It does **nothing** in the main server thread at all besides sending chat messages. It's not a hard concept to understand: two threads that don't interact with each other will not be able to break each other. DiscordSRV doesn't do anything to make the server hang in any way. Get some evidence to back up what you're saying and I can investigate it, but as it is now, you're making claims with no proof behind them.

Paste is removed so I can't give any actual assessment of my own.

**Comment by Curtis73** [comment]
I've done some more testing and it appears that my original result of DiscordSRV preventing load was inaccurate, and it's actually this plugin that was causing the issue -https://dev.bukkit.org/projects/serverlistmotd. I'm not sure how many people use it, but you might want to add it as an incompatibility or something.
Thanks for your support & time!
